### ISSUE TITLE
Game won't save and quit if i sell different items in from the submarine

### ISSUE THREAD

**Issue by hdnfbp** [Regalis11/Barotrauma#13090]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When i sell items to merchant directly from the submarine (3+ different items) i can't save and quit, when i try the borders blur (as if the server was closed) but the game is still going, i can't go through doors and if i use any item in my hands, all my inventory drops to the floor, this happens to all players on the server, even if i'm alone

I tested it on a dedicated server too, the same thing happens, there's nothing on the log or the console, i tried to validate the game file, initially 2 files were missing, the second time i tried all were validated, these files are from after the validation

![image](https://github.com/Regalis11/Barotrauma/assets/105054904/1ff21691-cb7a-4f74-aa05-89c84215ba8d)
![image](https://github.com/Regalis11/Barotrauma/assets/105054904/d5cba569-ea32-44c1-97f6-a8a126021425)
[shit.inc_2023-11-26_1017.txt](https://github.com/Regalis11/Barotrauma/files/13467133/shit.inc_2023-11-26_1017.txt)
![image](https://github.com/Regalis11/Barotrauma/assets/105054904/6856a7c7-017a-4ff7-a300-37aa4597befe)
[Save_bug.zip](https://github.com/Regalis11/Barotrauma/files/13467164/Save_bug.zip)

### Reproduction steps

1 - Download my save
2 - Sell at least 3 different items to the closest merchant (Ex: 1 bike horn, 1 electrical component, 1 revolver)
3 - Click save and quit
4 - Try going through doors
5 - Equip any aimable item from your inventory to your hand
6 - Aim and attack
7 - Watch yourself getting naked

Note: missions also can't be started after this point

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

Both in-game server and dedicated server

### Version

v1.1.19.3 (Treacherous Tides Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
There's no recent crash report
```

**Comment by hdnfbp** [comment]
New info, every time i open the save and validate the files, steam say there's 2 corrupted files

**Comment by Regalis11** [comment]
Thank you for the report! Testing this with the dedicated server, I got the following error message:
![image](https://github.com/Regalis11/Barotrauma/assets/14351070/1bf91405-c7eb-4fbf-bc2a-1a73069e8a01)

It seems this occurs when trying to sell components from inside a circuit box. We've got an existing ticket about the issue https://github.com/Regalis11/Barotrauma/issues/12968, and a fix coming up in the next patch, so I'll close the ticket as a duplicate. Sorry for the inconvenience!
